### ISSUE TITLE
Crash when not passing through "Connection: upgrade" headers

### ISSUE THREAD

**Issue by I-Al-Istannen** [mattermost/focalboard#112]
**Summary:**
The server (v.0.6.1, linux) crashes after a few page loads, refreshes and navigating the website *if nginx is configured incorrectly and does not pass through the connection upgrade header*.

**Steps to reproduce the behavior:**
1. Use an nginx config like:
```nginx
server {
    server_name "my domain";
    
    location / {
       proxy_set_header        Host $host:$server_port;
       proxy_set_header        X-Real-IP $remote_addr;
       proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
       proxy_set_header        X-Forwarded-Proto $scheme;
       proxy_set_header        X-Forwarded-Host $host;
#      proxy_set_header Upgrade $http_upgrade;
#      proxy_set_header Connection "upgrade";    

       proxy_pass              http://127.0.0.1:8000;
    }

    listen 80;
}
```
2. Refresh the page, click on random things, somehow provoke the right request
3. Observe a 502 (bad gateway) and the following log lines:
```
2021/03/18 20:16:48 DBType from commandline: sqlite3
2021/03/18 20:16:48 DBConfigString overriden from commandline
2021/03/18 20:16:48 connectDatabase
{"level":"info","ts":1616095008.5893445,"caller":"server/server.go:168","msg":"Server.Start"}
2021/03/18 20:16:48 http server started on :8000
2021/03/18 20:16:48 Starting unix socket server
2021/03/18 20:17:39 Single User: false
2021/03/18 20:17:39 Single User: false
2021/03/18 20:17:39 Single User: false
2021/03/18 20:17:39 Single User: false
2021/03/18 20:17:39 GetSubTree (2) blockID: a7bd568e-6b37-4c2b-8ca2-ef61a487eed4, 6 result(s)
2021/03/18 20:17:39 websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
systemd[315]: focalboard.service: Main process exited, code=exited, status=1/FAILURE
systemd[315]: focalboard.service: Failed with result 'exit-code'.
```

The last requests were:
```
GET /api/v1/blocks/a7bd568e-6b37-4c2b-8ca2-ef61a487eed4/subtree?l=2 HTTP/1.1" 200 # 200 OK
GET /ws/onchange HTTP/1.1" 502                                                    # 502 - Bad gateway (i.e. focalboard died)
```

**Expected behavior:**
The application should not crash just because it missed a header / websocket upgrade.

**Screenshots:**
Not applicable

**Platform:**
 - Firefox 86
 - Arch linux, kernel: 5.11.7-arch1-1

**Additional context:**
Not applicable

**Comment by jespino** [comment]
Totally agree, thanks for the report

**Comment by jespino** [comment]
This was fixed by @chenilim in the commit 0b479ff3
